crawl-ref/source/game-options.cc

```diff
@@ -66,6 +95,108 @@ bool read_bool(const string &field, bool def_value)
     return def_value;
 }
 
+class option_chooser : public Menu
+{
+public:
+    option_chooser(int _flags, GameOption *_caller)
+        : Menu(_flags), caller(_caller) { }
+    int pre_process(int key) override
+    {
+        if ('?' == key)
+        {
+            caller->show_help();
+            return static_cast<int>(CK_NO_KEY);
+        }
+        else
+            return key;
```

Comment:
DCSS generally uses early returns. (There are exceptions, because this project is ridiculous, but that's our modern style.)

```suggestion
        return key;
```

Reply:
I think I based that on EGP_Menu::pre_process(), which has more cases. I've changed it now.